webapp/src/ts/modules/messages/messages.component.ts

```diff
@@ -148,10 +156,21 @@ export class MessagesComponent implements OnInit, OnDestroy {
     this.updateActionBar();
   }
 
-  updateConversations({merge = false} = {}) {
-    return this.messageContactService
-      .getList()
-      .then((conversations = []) => {
+  updateConversations({ merge = false } = {}) {
+    return Promise
+      .all([ this.messageContactService.getList(), this.currentLevel ])
+      .then(([ conversations, currentLevel ]) => {
+        // Remove the lineage level that belongs to the offline logged-in user.
+        if (currentLevel) {
+          conversations?.forEach(conversation => {
+            if (conversation.lineage && conversation.lineage.length) {
+              conversation.lineage = conversation.lineage.filter(level => level);
+              if(conversation.lineage[conversation.lineage.length-1] === currentLevel){
+                conversation.lineage.pop();
+              }
+            }
+          });
+        }
```

Comment:
the issue here is that I try to avoid to have `conversation.lineage = undefined ` in the output, when lineage data is missing from conversation because it's messing previous unit tests up.

Reply:
@elvisdorkenoo  you can exit early
```
if (conversation.lineage?.length) {
  return;
}
conversation.lineage = conversation.lineage?.filter(level => level);
.... continue with the code ....
```

Reply:
yeah tha's flatter, thanks.